src/document.c

```diff
@@ -94,10 +94,18 @@ RSAddDocumentCtx *NewAddDocumentCtx(IndexSpec *sp, Document *base) {
       aCtx->fspecs[i] = *fs;
       if (FieldSpec_IsSortable(fs) && aCtx->sv == NULL) {
         aCtx->sv = NewSortingVector(sp->sortables->len);
+        // mark sortable fields to be updated in the state flags
+        aCtx->stateFlags |= ACTX_F_SORTABLES;
```

Comment:
You're never using this flag.

Reply:
it should be used to avoid looping on all the fields in partial mode, if there aren't any sortables.